### PR TITLE
build: enable typescript for cypress

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Follow up #10161 :

- Enable TypeScript for Cypress
- Convert support files to TS
- Convert an example test (dashboard/filter.test.js) to TS
- Reenable Prefer TS check for `cypress-base`

<strike>Currently `cypress-base` is now using a newer version of TypeScript than `superset-frontend`. Will file another PR to upgrade later.</strike>

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

All Cypress tests should still be valid and pass.

cc @etr2460

### CODE REVIEW

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts [hunk 1]
```diff
@@ -72,7 +85,7 @@ describe('Dashboard filter', () => {
 
     cy.get('.Select__control input[type=text]')
       .first()
-      .focus({ force: true })
```
Comment (ktmud) [inline superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts:75]: Type check helped by notifying me this is not a valid API for `focus`.

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts [hunk 2]
```diff
@@ -32,13 +44,14 @@ describe('Dashboard filter', () => {
...
+      const bootstrapData = app.data('bootstrap');
+      const dashboard = bootstrapData.dashboard_data as DashboardData;
       const sliceIds = dashboard.slices.map(slice => slice.slice_id);
-      filterId = dashboard.slices.find(
-        slice => slice.form_data.viz_type === 'filter_box',
-      ).slice_id;
+      filterId =
+        dashboard.slices.find(
+          slice => slice.form_data.viz_type === 'filter_box',
+        )?.slice_id || 0;
```
Comment (ktmud) [inline superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts:54]: Fallback to 0 is easier than running assertions on a potential `undefined`.

**Comment by etr2460** [comment]
Looks like CI is still breaking a bit, will review once fixed (i think you're missing installing an import)

File: superset-frontend/cypress-base/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["cypress", "@typescript-eslint"],
+  "extends": [
+    "plugin:@typescript-eslint/recommended",
+    "plugin:cypress/recommended"
+  ],
+  "rules": {
+    "import/no-unresolved": 0,
```
Comment (ktmud) [inline superset-frontend/cypress-base/.eslintrc:9]: Some Cypress modules might not be available if you run `eslint` in the parent folder before `npm install` in `cypress-base` (which is what our CI does).

**Comment by ktmud** [comment]
@etr2460 Fixed the tests and this is now ready for review.

**Review by etr2460 — APPROVED** [review]
lgtm, thanks for taking the time to do this!

**Pull request merged**